Simplify list descriptions and add release process (#97)
mozilla-services/shavar-prod-lists: README.md

This repo serves as a staging area for
[shavar](https://github.com/mozilla-services/shavar) /
[tracking protection](https://wiki.mozilla.org/Security/Tracking_protection)
lists prior to
[production deployment to Firefox](https://mana.mozilla.org/wiki/display/SVCOPS/Shavar+-+aka+Mozilla's+Tracking+Protection).
This repo gives Mozilla a chance to manually review all updates before they go
live, a fail-safe to prevent accidental deployment of a list that could break
Firefox.

These lists are processed and transformed and sent to Firefox via
[Shavar](https://mana.mozilla.org/wiki/display/SVCOPS/Shavar+-+aka+Mozilla's+Tracking+Protection).

## Disconnect's Lists
Firefox's Enhanced Tracking Protection features rely on lists of trackers
maintained by [Disconnect](https://disconnect.me/trackerprotection).
Mozilla does not maintain these lists. As such, we will close all issues and
pull requests related to making changes to the list contents. These issues
should be reported to Disconnect.

#### `disconnect-blacklist.json`
A version controlled copy of Disconnect's
[list of trackers](https://github.com/disconnectme/disconnect-tracking-protection/blob/master/services.json).

> This blacklist is the core of tracking protection in Firefox.

Reviewer: ```suggestion
This blocklist is the core of tracking protection in Firefox.
```

This remapping occurs at the time of list creation, so the Social, Analytics,
and Advertising lists consumed by Firefox will contain these domains.

Firefox consumes the list as follows:
* **Tracking**: anything in the Advertising, Analytics, Social, Content, or
    Disconnect category. Firefox ships two versions of the tracking lists: the
    "Level 1" list, which excludes the "Content" category, and the
    "Level 2" list which includes the "Content" category.
* **Cryptomining**: anything in the Cryptomining category
* **Fingerprinting**: anything in the Fingerprinting category. By default,
    ETP's fingerprinting blocking only blocks _Tracking Fingerprinters_, that
    is domains which appear in both the Fingerprinting category and one of the
    Tracking categories.

#### `disconnect-entitylist.json`

A version controlled copy of Disconnect's
[list of entities](https://github.com/disconnectme/disconnect-tracking-protection/blob/master/entities.json).
ETP classifies a resource as a tracking resource when it is present on

> blocklist and loaded as a third-party. The Entity list is used to whitelist

Reviewer: ```suggestion
blocklist and loaded as a third-party. The Entity list is used to allow
```

third-party subresources that are wholly owned by the same company that owns
the top-level website that the user is visiting. For example, if abcd.com owns

> efgh.com and efgh.com is on the blacklist, it will not be blocked on abcd.com.

Reviewer: ```suggestion
efgh.com and efgh.com is on the blocklist, it will not be blocked on abcd.com.
```

Instead, efgh.com will be treated as first party on abcd.com, since the same

> company owns both. But since efgh.com is on the blacklist it will be blocked on

Reviewer: ```suggestion
company owns both. But since efgh.com is on the blocklist it will be blocked on
```

other third-party domains that are not all owned by the same parent company.

## Other lists

In addition, Mozilla maintains a several lists for Firefox-specific features
and experiments. The lists currently in active use are:
* `social-tracking-protection-blacklist.json`: a subset of social trackers from
    Disconnect's blocklist. This list is used to identify "social media"
    trackers within Firefox's UI. All of the origins on this list should also

>     be included in Disconnect's `disconnect-blacklist.json` list.

Reviewer: ```suggestion
    be included in "Social" category of Disconnect's `disconnect-blacklist.json` list.
```

Author: I'm not sure this is true right now, and I don't think we intend to restrict ourselves to the Social category. Have you heard otherwise?

Reviewer: Oh, I was under the impression that all the domains on this is already listed under the "Social" category in the Disconnect's lists. Isn't that what you meant on the [sentences](https://github.com/mozilla-services/shavar-prod-lists/pull/97/files#diff-04c6e90faac2675aa89e2176d2eec7d8R61-R62) before? I was reiterating that.

Author: I've updated this to "a subset of trackers ..."

* `google-mapping.json`: a legacy entity remapping file used by Firefox iOS.

>     This list should not be used be any new products. Instead, the

Reviewer: ```suggestion
    This list should not be used in any new products. Instead, the
```

changes from Disconnect as soon as they are available. When a new version of
Firefox is released, we will also release a new version of the list that
corresponds to the version of Firefox moving from Nightly (main branch) --> Beta
(versioned branch). That version of the list will ride the trains along with its
respective Firefox version. Releases older than Firefox 69 use the 69 version of
the blocklist.

This means that all changes will be tested for at least the full beta cycle and
part of the Nightly cycle. We may choose to shorten the testing cycle in the
future.

There are three possible exceptions to this process:
1. **Fast-tracked changes** which are deployed immediately to all channels
2. **Temporary exceptions** which are deployed using Remote Settings
3. **List freezes** for when we’d like to test changes for a longer duration.
   These are tracked in Github issues on this repository.

#### Fast-tracked changes

We will fast track breakage-related updates or policy-related updates, both
of which may only be done by Disconnect. Fast-tracked changes should have
minimal, if any, risk of breakage.

Changes that may be fast-tracked include:

> * Deleting a domain from the list.

Reviewer: ```suggestion
* Deleting a domain from the blocklist.
```

Reviewer: Or will this be applied to both entity and block lists?

Author: It can be both, but only in the case where a domain is deleted from the blocklist.

Author: I'll update the terminology here to `Deleting a domain from the blocklist and it's respective domains from the entitylist.`

* Adding new domains to the entity list.
* Replacing a domain currently on the list with a new domain at the request of

>   the entity.

Reviewer: Can you clarify what you mean by "entity"? Is this the company that is requesting this change and has been approved by Disconnect?

Author: Yes, I'll update this.

* Moving a domain between list categories of the same feature.

As soon as Disconnect makes changes of this type we will merge
them into each versioned list and deploy them across all channels.

#### Temporary exceptions

If we discover breakage that we'd like to fix through Firefox code changes, we

> may temporarily whitelist the relevant domains while we work on the fixes. To

Reviewer: ```suggestion
may temporarily add the relevant domains to the entity list in Remote Settings while we work on the fixes. To
```

Author: This is not true, but I'll update the whitelist terminology. This is not related to the entity list.

prevent these from becoming permanent exceptions, a deadline must be provided by

> the engineer at the time of whitelisting, with a maximum deadline of 18 months.

Reviewer: ```suggestion
the engineer at the time of adding the domain to the entity list, with a maximum deadline of 18 months.
```

Author: Same here
